gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -617,14 +618,28 @@ gis_network_page_constructed (GObject *object)
       if (!nm_device_get_managed (device))
         continue;
 
-      if (nm_device_get_device_type (device) == NM_DEVICE_TYPE_WIFI) {
+      switch (nm_device_get_device_type (device)) {
+      case NM_DEVICE_TYPE_ETHERNET:
+        has_ethernet_connection = TRUE;
+        priv->nm_device = g_object_ref (device);
```

Comment:
This only works if you have just one NMDevice, but this is a loop that handles multiple NMDevices. So in the common case of ethernet plus wi-fi device, you leak whichever one gets processed first. At minimum you need a g_clear_object in both cases, but probably you don't really need to set priv->nm_device if it's Ethernet at all, right...?